### PR TITLE
Add MCSource and MCSourceService

### PR DESCRIPTION
**Pull request by Chenyuyao**
Right now both of them are temporary solution
They are just for hooking and debuting purpose
They are to be redesigned very shortly
Add source and needParse Property to MCParsedRSSItem
Modify MCRSSService to enable actual fetching

### CODE REVIEW

File: motcha/DataModel/MCParsedRSSItem.m [hunk 1]
```diff
@@ -19,10 +19,15 @@ - (instancetype)initWithTitle:(NSString *)title
...
-  
-  //TODO:extract RSS source
+
   return self;
 }
 
+- (void) addSource:(NSString *)source
+         needParse:(BOOL)needParse {
+  _source = source;
+  needParse = needParse;
```
Comment (franklixuefei) [inline motcha/DataModel/MCParsedRSSItem.m]: should be _needParse = needParse; ?

File: motcha/Network/MCSourceService.m [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+- (NSArray *)getSourceWithCategory:(NSString *)category {
+  NSArray * result = [_sources objectForKey:category];
+  if (result == nil) {
+    result = [_sources objectForKey:@"technology"];
+  }
+  return result;
+}
+
+@end
```
Comment (franklixuefei) [inline motcha/Network/MCSourceService.m]: 回个车

File: motcha/Network/MCSourceService.m [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                           needParse:NO
+                         fullTextale:YES];
+  [_sources setValue:@[artsSource] forKey:@"arts"];
+  
+}
+
+- (NSArray *)getSourceWithCategory:(NSString *)category {
+  NSArray * result = [_sources objectForKey:category];
+  if (result == nil) {
+    result = [_sources objectForKey:@"technology"];
```
Comment (franklixuefei) [inline motcha/Network/MCSourceService.m]: static NSString *kDefaultCategory = @"technology";

@implementation ....
...

result = [_sources objectForKey:kDefaultCategory];

感觉更readable一些

File: motcha/Network/MCSourceService.m [hunk 4]
```diff
@@ -0,0 +1,49 @@
+#import "MCSourceService.h"
+#import "MCSource.h"
+
+@implementation MCSourceService
+
```
Comment (franklixuefei) [inline motcha/Network/MCSourceService.m]: 这个service可以加一个+sharedinstance 和其它一样

File: motcha/Network/MCSourceService.h [hunk 5]
```diff
@@ -0,0 +1,13 @@
+#import <Foundation/Foundation.h>
+
+//A service to return manage MCSource
+//Right now we hardcode every MCSource object
+//TODO: Should change to extracting source info from CORE DATA
+@interface MCSourceService : NSObject
+
+@property(nonatomic, readonly) NSDictionary *sources;
```
Comment (franklixuefei) [inline motcha/Network/MCSourceService.h]: comment what are the types of the key and the value

File: motcha/Network/MCSourceService.h [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+- (NSArray *) getSourceWithCategory:(NSString *) category;
```
Comment (franklixuefei) [inline motcha/Network/MCSourceService.h]: 小细节 这里可以rename 一下 getSourcesWithCategory

File: motcha/Network/MCRSSService.m [hunk 7]
```diff
@@ -12,15 +17,71 @@ + (MCRSSService *)sharedInstance {
...
+    [parser setShouldResolveExternalEntities:NO];
+    [parser parse];
+    
+    tmpItemArray = [parserOperation feeds];
+    
+    //Reprocess MCParsedRSSItem
+    NSMutableIndexSet *discardedItems = [NSMutableIndexSet indexSet];
+    NSUInteger index = 0;
+    
+    for (MCParsedRSSItem * item in tmpItemArray ) {
```
Comment (franklixuefei) [inline motcha/Network/MCRSSService.m]: 小问题 我觉得这里可以改成
for (NSUInteger index = 0; index < [tmpItemArray count]; ++index) {
...
}

这样直接用index感觉自然一点

File: motcha/Network/MCRSSService.m [hunk 8]
```diff
@@ -12,15 +17,71 @@ + (MCRSSService *)sharedInstance {
...
-  [parser parse];
+- (NSMutableArray *)fetchRSSWithCategory:(NSString *)category
+                                   since:(NSDate *)since {
+  //get all MCSource related to category
+  MCSourceService * sourceService = [[MCSourceService alloc] init];
+  NSArray * sourceArray;
+  NSMutableArray * rssItemArray = [NSMutableArray array];
+  
+  [sourceService hardCodeSource];
+  sourceArray = [sourceService getSourceWithCategory:category];
```
Comment (franklixuefei) [inline motcha/Network/MCRSSService.m]: line 24 和这行合并

File: motcha/DataModel/MCParsedRSSItem.m [hunk 9]
```diff
@@ -19,10 +19,17 @@ - (instancetype)initWithTitle:(NSString *)title
     [dateFormat setDateFormat:@"EEE, dd MMM yyyy HH:mm:ss"];
     _pubDate = [dateFormat dateFromString:pubDate];    
   }
-  
-  //TODO:extract RSS source
+
   return self;
 }
 
+- (void) addSource:(NSString *)source
```
Comment (franklixuefei) [inline motcha/DataModel/MCParsedRSSItem.m]: rename成setSource:category:needParse: 感觉好一些

File: motcha/Network/MCSourceService.h (hunk 5 above)
Comment (Chenyuyao) [inline motcha/Network/MCSourceService.h]: 这两个文件我以后会重新design 咱先merge debug吧

**Comment by franklixuefei** [comment]
没什么问题 咱们先merge debug吧 等ok了再改

**Pull request merged**